sdk/confidentialledger/Azure.Security.ConfidentialLedger/tests/ConfidentialLedgerClientTests.cs

```diff
@@ -54,9 +55,9 @@ public void FailedTransaction()
                             })
                     }
                     ));
-
+            var operation = await client.PostLedgerEntryAsync(WaitUntil.Started, RequestContent.Create(new { contents = "test" }), null, default);
```

Comment:
@christothes I changed the implementation of the test to pass WaitUntil.Started. Otherwise InstrumentResultInterceptor would swap out Completed for Started and then after invoking the method would call waitForCompletion(as per the changes in this PR), but we would then hit a failure in the Diagnostic Scope validation because the scope would not be marked as failed. This points to a more general issue where this swapping mechanism doesn't play nice with tests that expect a failure. Since this was the only such test, and it is not actually even a live test, I figured it was probably fine to just change the test rather than trying to figure out a way to support the test in its original implementation.